packages/arb-token-bridge-ui/src/components/syncers/useBalanceUpdater.tsx

```diff
@@ -1,24 +1,65 @@
-import { useInterval, useLatest } from 'react-use'
+import { useMemo } from 'react'
+import { useInterval, useLatest, useDebounce } from 'react-use'
 import { useAccount } from 'wagmi'
 
 import { useAppState } from '../../state'
 import { isTokenNativeUSDC } from '../../util/TokenUtils'
 import { useSelectedToken } from '../../hooks/useSelectedToken'
 import { useUpdateUsdcBalances } from '../../hooks/CCTP/useUpdateUsdcBalances'
+import { useBalances } from '../../hooks/useBalances'
 
 // Updates all balances periodically
 export function useBalanceUpdater() {
   const {
     app: { arbTokenBridge }
   } = useAppState()
+
   const [selectedToken] = useSelectedToken()
   const { address: walletAddress } = useAccount()
   const latestTokenBridge = useLatest(arbTokenBridge)
+  const { bridgeTokens } = latestTokenBridge.current
+
+  const { updateErc20ParentBalances, updateErc20ChildBalances } = useBalances({
+    parentWalletAddress: walletAddress,
+    childWalletAddress: walletAddress
+  })
 
   const { updateUsdcBalances } = useUpdateUsdcBalances({
     walletAddress
   })
 
+  const parentErc20Addresses = useMemo(
+    () =>
+      (bridgeTokens
+        ? Object.keys(bridgeTokens ?? {}).map(key => bridgeTokens[key]?.address)
+        : []
+      ).filter(address => address !== undefined),
+    [bridgeTokens]
+  )
+
+  const childErc20Addresses = useMemo(
+    () =>
+      (bridgeTokens
+        ? Object.keys(bridgeTokens ?? {}).map(
+            key => bridgeTokens[key]?.l2Address
+          )
+        : []
+      ).filter(address => address !== undefined),
+    [bridgeTokens]
+  )
+
+  // when the wallet address changes / wallet is freshly connected, fetch and update balances of all tokens held by user
```

Comment:
Earlier, the bridge assumed that the wallet address would always be present. So it used to load app (which included fetching balances) only when wallet used to connect. But going forward, we load the app first, and expect the balances to be fetched AFTER the user connects the wallet.

This piece of functionality (fetching balances when user connected with wallet) was missing in our Bridge. Fixed here.